src/envoy/utils/utils.cc

```diff
@@ -34,6 +34,29 @@ const std::string kPerHostMetadataKey("istio");
 // Attribute field for per-host data override
 const std::string kMetadataDestinationUID("uid");
 
+bool getCertSAN(const Network::Connection* connection, bool peer,
+                std::string* principal) {
+  if (connection) {
+    Ssl::ConnectionInfo* ssl =
+        const_cast<Ssl::ConnectionInfo*>(connection->ssl());
+    if (ssl != nullptr) {
+      const std::vector<std::string> sans =
+          (peer ? ssl->uriSanPeerCertificate() : ssl->uriSanLocalCertificate());
+      if (sans.empty()) {
+        // empty result is not allowed.
+        return false;
+      }
+      *principal = sans[0];
+      return true;
+    }
+  }
+  return false;
+}
+
+bool hasSPIFFEPrefix(const std::string& san) {
```

Comment:
Probably you are implementing absl::StartsWith

Reply:
Changed to use absl::StartsWith.